include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -369,14 +369,7 @@ class open_addressing_ref_impl {
       auto const window_slots = storage_ref_[*probing_iter];
 
       for (auto& slot_content : window_slots) {
-        auto const eq_res =
-          this->predicate_.operator()<is_insert::YES>(key, this->extract_key(slot_content));
-
-        if constexpr (not allows_duplicates) {
-          // If the key is already in the container, return false
-          if (eq_res == detail::equal_result::EQUAL) { return false; }
-        }
-        if (eq_res == detail::equal_result::AVAILABLE) {
+        if constexpr (is_perfect_hashing<decltype(probing_scheme_)>::value) {
```

Comment:
I would like to check if the function `is_perfect` exists, which you are doing here, and *also* check that `is_perfect()` returns `true`.  Because we want to allow someone to define the function as false.  Like, say you are inheriting from a class that defines the function.  You can't easily undefine it.  So you might choose to override the inheritance with a version that returns false.

(Technically, you could also do `private` inheritance so that the `is_perfect` function doesn't get defined in the class that extends but that also has other side-effects that the user might not want!)